src/app/Rules/CreateRule.tsx

```diff
@@ -62,6 +64,7 @@ const Comp = () => {
   const [description, setDescription] = React.useState('');
   const [matchExpression, setMatchExpression] = React.useState('');
   const [matchExpressionValid, setMatchExpressionValid] = React.useState(ValidatedOptions.default);
+  const [target, setTarget] = React.useState(undefined as Target | undefined);
```

Comment:
I think you could probably default this to the `NO_TARGET` special value, rather than `undefined`. If so then I think that'd be preferred since you don't need to worry about null/undefined dereferences in that case.